app/src/main/java/com/github/libretube/SearchFragment.kt

```diff
@@ -143,4 +176,36 @@ class SearchFragment : Fragment() {
         super.onStop()
         hideKeyboard()
     }
+
+    private fun addtohistory(query: String) {
+        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(requireContext())
+
+        var splited_history = getHistory()
+
+        if (query == splited_history.get(splited_history.size - 1)) {
+            return
+        } else if (query == "") {
+            return
+        } else {
+            splited_history = splited_history + query
+        }
+
+
+
+        if (splited_history.size > 10) {
+            splited_history = splited_history.takeLast(10)
+        }
+
+
+        sharedPreferences.edit().putString("search_history", splited_history.joinToString("|"))
```

Comment:
What if I search for something that contains `|`?

Reply:
Yeah you are right
Any suggestions?

Reply:
To write the objects: (I haven't tested if this works)
```kotlin
        val baos = ByteArrayOutputStream()
        ObjectOutputStream(baos).use {
            it.writeObject(arrayOf("1", "2", "3", "4", "5"))
        }
        baos.use {
            val encoded = Base64.encodeToString(it.toByteArray(), Base64.DEFAULT)
        }
```

Reply:
Will test...
Thx